woocommerce/woocommerce
# Multichannel Marketing - Changes to the marketing classes

### All Submissions:






### Changes proposed in this Pull Request:



This PR implements the following changes to the core library (initially introduced in https://github.com/woocommerce/woocommerce/pull/35099):

- Allow third-party channel registration without checking the extension's slug against a predetermined list. Any extension can now register as a marketing channel if they use the classes introduced in #35099. Previously, we checked if the extension's slug matches one of the values returned by the `GET https://woocommerce.com/wp-json/wccom/marketing-tab/1.2/recommendations.json` API before allowing them to register as a marketing channel.
- Change the name of the method `MarketingChannelInterface::get_error_no` to `get_error_count`. The reasoning for this change was that `get_error_count` describes more clearly what the method returns. (reference: pe2C5g-zb-p2)
- Revert the changes made to the `InstalledExtensions.php` class and its usage. This is a legacy class, and it's best to keep it intact since we are introducing the new marketing dashboard as an opt-in feature. The new feature will be implemented as a new API in https://github.com/woocommerce/woocommerce/issues/34556

Closes #35956.





### How to test the changes in this Pull Request:



Some of the changes in this PR can not be tested yet because they still require an API to be implemented in https://github.com/woocommerce/woocommerce/issues/34556 and a front-end as described in https://github.com/woocommerce/woocommerce/issues/34903.

We can test if the changes to `InstalledExtensions` have reverted the Marketing page to its original state (before #35099):

1. Install some marketing extensions returned by the `GET https://woocommerce.com/wp-json/wccom/marketing-tab/1.2/recommendations.json` API. For example, AutomateWoo, Facebook for WooCommerce, or Pinterest for WooCommerce.
2. Go to Marketing > Overview and confirm that all the installed extensions appear on the "Installed Extensions" list.



### Other information:




### FOR PR REVIEWER ONLY:
